course05/homework01/joel/string.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <ostream>
+#include <string>
+
+class String
+{
+public:
+	String(const char s[]) : mSize(std::strlen(s)), mContents(std::make_unique<char[]>(mSize))
+	{
+		std::strcpy(mContents.get(), s);
```

Comment:
This might overflow, you should check the length and throw before copying. I wouldnt use strcpy, it's a C function exported to the std:: NS - we would always prefer algorithms like std::copy

Reply:
In general the convention would to pass `const char* str, size_t len`, not just a pointer or array